runtime/codert_vm/xnathelp.m4

```diff
@@ -972,23 +972,12 @@ START_PROC(jitReferenceArrayCopy)
 	pop uword ptr J9TR_VMThread_jitReturnAddress[_rbp]
 	SWITCH_TO_C_STACK
 	SAVE_ALL_REGS
-
-	mov    uword ptr [_rbp+J9TR_VMThread_floatTemp3], _rsi
-	mov    uword ptr [_rbp+J9TR_VMThread_floatTemp4], _rdi
-ifdef({ASM_J9VM_ENV_DATA64},{
-ifdef({WIN32},{
-	mov    _rdx, _rcx
-	mov    _rcx, _rbp
-},{
-	mov    _rsi, _rcx
-	mov    _rdi, _rbp
-})
-},{
-	mov    _rdx, _rcx
-	mov    _rcx, _rbp
-})
-	call   FASTCALL_SYMBOL(impl_jitReferenceArrayCopy,2)
-	test   _rax, _rax
+	mov uword ptr J9TR_VMThread_floatTemp3[_rbp],_rsi
+	mov uword ptr J9TR_VMThread_floatTemp4[_rbp],_rdi
+	mov PARM_REG(2),_rcx
```

Comment:
This is great, but I'm a little concerned that since we don't know here what arg1 and arg2 are, it's possible we might be smashing registers unintentionally (I have no good solution for this).

As with the readBarrier code, you can probably get away with only saving C volatile registers here, not all registers. Also, I don't think you need the return address pop/push.

Reply:
See above re: C volatile regs and return address.

Reply:
Good suggestion, updated.